File: apps/client/lib/data/api/sdkproto/v1/biumind_ext.dart
// BiuMind 业务扩展 —— CreateSessionReq / Session / EnvironmentInfo + Mode / WorkerKind 常量。

import 'package:json_annotation/json_annotation.dart';

part 'biumind_ext.g.dart';

class Mode {
  static const chat = 'chat';
  static const agent = 'agent';
  static const task = 'task';
}

class WorkerKind {
  static const biuDaemon = 'biu_daemon';
  static const biuCli = 'biu_cli';
  static const runtime = 'runtime';
}

class EnvState {
  static const online = 'online';
  static const offline = 'offline';
  static const draining = 'draining';
}

@JsonSerializable(includeIfNull: false, anyMap: true)
class CreateSessionReq {
  final String mode;
  @JsonKey(name: 'environment_id')
  final String? environmentId;
  @JsonKey(name: 'thread_id')
  final String? threadId;
  final String? model;
  @JsonKey(name: 'system_prompt')
  final String? systemPrompt;
  @JsonKey(name: 'biumind_attachment_ids')
  final List<String>? biumindAttachmentIds;

  CreateSessionReq({
    required this.mode,
    this.environmentId,
    this.threadId,
    this.model,
    this.systemPrompt,
    this.biumindAttachmentIds,
  });

  factory CreateSessionReq.fromJson(Map<String, dynamic> json) =>
      _$CreateSessionReqFromJson(json);
  Map<String, dynamic> toJson() => _$CreateSessionReqToJson(this);
}

@JsonSerializable(includeIfNull: false, anyMap: true)
class Session {
  @JsonKey(name: 'session_id')
  final String sessionId;
  @JsonKey(name: 'session_token')
  final String sessionToken;
  final String mode;
  @JsonKey(name: 'server_seq_start')
  final int serverSeqStart;
  @JsonKey(name: 'jetstream_subject_in')
  final String? jetstreamSubjectIn;
  @JsonKey(name: 'jetstream_subject_out')
  final String? jetstreamSubjectOut;
  @JsonKey(name: 'environment_id')
  final String? environmentId;
  @JsonKey(name: 'thread_id')
  final String? threadId;

  Session({
    required this.sessionId,
    required this.sessionToken,
    required this.mode,
    required this.serverSeqStart,
    this.jetstreamSubjectIn,
    this.jetstreamSubjectOut,
    this.environmentId,
    this.threadId,
  });

  factory Session.fromJson(Map<String, dynamic> json) =>
      _$SessionFromJson(json);
  Map<String, dynamic> toJson() => _$SessionToJson(this);
}

@JsonSerializable(includeIfNull: false, anyMap: true)
class EnvironmentInfo {
  @JsonKey(name: 'environment_id')
  final String environmentId;
  @JsonKey(name: 'user_id')
  final String? userId;
  @JsonKey(name: 'worker_kind')
  final String workerKind;
  @JsonKey(name: 'machine_name')
  final String machineName;
  @JsonKey(name: 'os_arch')
  final String? osArch;
  @JsonKey(name: 'git_info')
  final Map<String, dynamic>? gitInfo;
  final List<String>? capabilities;
  final String state;
  @JsonKey(name: 'last_seen_at')
  final int? lastSeenAt;

  EnvironmentInfo({
    required this.environmentId,
    this.userId,
    required this.workerKind,
    required this.machineName,
    this.osArch,
    this.gitInfo,
    this.capabilities,
    required this.state,
    this.lastSeenAt,
  });

  factory EnvironmentInfo.fromJson(Map<String, dynamic> json) =>
      _$EnvironmentInfoFromJson(json);
  Map<String, dynamic> toJson() => _$EnvironmentInfoToJson(this);
}

File: packages/go-sdk/biu/sdkproto/v1/biumind_ext.go
package sdkproto

import "encoding/json"

// Mode 字面量 —— 跟 schema biumind_ext.json 的 Mode enum 对齐。
const (
	ModeChat  = "chat"
	ModeAgent = "agent"
	ModeTask  = "task"
)

// Worker 类型字面量。
const (
	WorkerKindBiuDaemon = "biu_daemon"
	WorkerKindBiuCLI    = "biu_cli"
	WorkerKindRuntime   = "runtime"
)

// Environment state 字面量。
const (
	EnvStateOnline   = "online"
	EnvStateOffline  = "offline"
	EnvStateDraining = "draining"
)

// CreateSessionReq 是 POST /v1/code/sessions 的请求体。
type CreateSessionReq struct {
	Mode                 string   `json:"mode"`
	EnvironmentID        string   `json:"environment_id,omitempty"`
	ThreadID             string   `json:"thread_id,omitempty"`
	Model                string   `json:"model,omitempty"`
	SystemPrompt         string   `json:"system_prompt,omitempty"`
	BiumindAttachmentIDs []string `json:"biumind_attachment_ids,omitempty"`
}

// Session 是 POST /v1/code/sessions 的响应体。
type Session struct {
	SessionID           string `json:"session_id"`
	SessionToken        string `json:"session_token"`
	Mode                string `json:"mode"`
	ServerSeqStart      int64  `json:"server_seq_start"`
	JetstreamSubjectIn  string `json:"jetstream_subject_in,omitempty"`
	JetstreamSubjectOut string `json:"jetstream_subject_out,omitempty"`
	EnvironmentID       string `json:"environment_id,omitempty"`
	ThreadID            string `json:"thread_id,omitempty"`
}

// EnvironmentInfo 来自 GET /v1/agent/environments。
type EnvironmentInfo struct {
	EnvironmentID string          `json:"environment_id"`
	UserID        string          `json:"user_id,omitempty"`
	WorkerKind    string          `json:"worker_kind"`
	MachineName   string          `json:"machine_name"`
	OsArch        string          `json:"os_arch,omitempty"`
	GitInfo       json.RawMessage `json:"git_info,omitempty"`
	Capabilities  []string        `json:"capabilities,omitempty"`
	State         string          `json:"state"`
	LastSeenAt    int64           `json:"last_seen_at,omitempty"`
}
